assets/js/modules/analytics/common/account-create.js

```diff
@@ -0,0 +1,180 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import { each } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useState, Fragment, useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Button from '../../../components/button';
+import ProgressBar from '../../../components/progress-bar';
+import { trackEvent } from '../../../util';
+import TimezoneSelect from './timzezone-select';
+import AccountField from './account-field';
+import PropertyField from './property-field';
+import ProfileField from './profile-field';
+import { STORE_NAME } from '../datastore/constants';
+import { countries } from './countries';
+import { STORE_NAME as CORE_STORE_NAME } from '../../../googlesitekit/datastore/site/constants';
+import Data from 'googlesitekit-data';
+const { useDispatch, useSelect } = Data;
+
+// Recursively search thru countries and their timezones to find a match for country/timezone.
+const timezoneInCountries = ( timezone ) => {
+	let matched = false;
+	each( countries.default.country, ( country ) => {
+		const timezoneMatch = country.timeZone.find( ( tz ) => tz.timeZoneId === timezone );
+		if ( timezoneMatch ) {
+			matched = true;
+		}
+	} );
+	return matched;
+};
+let timezoneChecked = false;
+
+const AccountCreate = () => {
+	const siteURL = useSelect( ( select ) => select( CORE_STORE_NAME ).getReferenceSiteURL() );
+	const siteName = useSelect( ( select ) => select( CORE_STORE_NAME ).getSiteName() );
+	let tz = useSelect( ( select ) => select( CORE_STORE_NAME ).getTimezone() );
+
+	// Check timezone on initial load: fall back to the browser timezone if the WordPress timezone was not found.
+	if ( ! timezoneChecked && ! timezoneInCountries( tz ) ) {
+		tz = Intl.DateTimeFormat().resolvedOptions().timeZone;
+	}
+	timezoneChecked = true;
+	const url = new URL( siteURL );
+	const { createAccount } = useDispatch( STORE_NAME );
+
+	const isDoingCreateAccount = useSelect( ( select ) => select( STORE_NAME ).isDoingCreateAccount() );
+
+	const accountTicketTermsOfServiceURL = useSelect( ( select ) => select( STORE_NAME ).getAccountTicketTermsOfServiceURL() );
+	const [ isNavigating, setIsNavigating ] = useState( false );
+	const handleSubmit = async function( accountName, propertyName, profileName, timezone ) {
+		trackEvent( 'analytics_setup', 'new_account_setup_clicked' );
+		setIsNavigating( true );
+		const result = await createAccount( {
+			accountName,
+			propertyName,
+			profileName,
+			timezone,
+		} );
+
+		if ( result.error ) {
+			setIsNavigating( false ); // Silently fail for server errors.
+		}
```

Comment:
This can be simplified a bit as the progress bar is already shown if doing create account. `setIsNavigating( true )` is only needed if create account was successful.

Reply:
simplified in https://github.com/google/site-kit-wp/pull/1389/commits/1959e34b9e88cfe463f38790f7de7899b6ac9c76